src/main/kotlin/com/ampnet/userservice/grpc/GrpcUserServer.kt

```diff
@@ -133,12 +140,37 @@ class GrpcUserServer(
         responseObserver.onCompleted()
     }
 
-    private fun getRole(role: SetRoleRequest.Role): UserRole =
+    override fun getAllActiveUsers(
+        request: GetActiveUsersRequest,
+        responseObserver: StreamObserver<UsersExtendedResponse>
+    ) {
+        logger.debug { "Received gRPC request getAllActiveUsers for coop: ${request.coop}" }
+        try {
+            val coop = coopService.getCoopByIdentifier(request.coop)
+                ?: throw ResourceNotFoundException(ErrorCode.COOP_MISSING, "Missing coop: ${request.coop} on platform")
+            val users = userRepository.findAllByCoopAndUserInfoUuidIsNotNull(request.coop)
+                .associateBy { it.userInfoUuid }
+            val usersWithExtendedInfo = userInfoRepository.findAllByCoop(request.coop).map { userInfo ->
+                users[userInfo.uuid]?.let { user -> buildUserExtendedResponse(user, userInfo) }
+            }
+            val response = UsersExtendedResponse.newBuilder()
+                .addAllUsers(usersWithExtendedInfo)
+                .setCoop(buildCoopResponse(coop))
+                .build()
+            responseObserver.onNext(response)
+            responseObserver.onCompleted()
+        } catch (ex: ResourceNotFoundException) {
+            logger.warn(ex) { "Could not get coop: ${request.coop}" }
+            responseObserver.onError(ex)
+        }
```

Comment:
Is this exception throw only from line 150? If that's true, it would be cleaner to call `onError` just form that place. This way it's not clear if the other repository calls can throw `ResourceNotFoundException`.